Remove `consultations` module

andreslucena opened the pull request:
<!--
NOTE: We are in the middle of a big redesign of the frontend.
That could mean that your PR will not get through on the next few months.
Please see https://github.com/decidim/decidim/discussions/9512
-->

#### :tophat: What? Why?

Since January 2021, we've decided that Consultations no longer be maintained, as we were working in the alternative: `decidim-elections`.

We've let people know that this is the case [through the documentation](https://docs.decidim.org/en/develop/admin/spaces/consultations.html) and also by a callout (see #7095) available from version v0.24.0. 

The time has come for it to die, as in the future v0.28, it was not even redesigned, so this PR removes it. 

#### :pushpin: Related Issues
 
- Fixes #11148

#### Testing

All the suite should be green. 
There should not be any more reference to `decidim-consultations` or consultations in the code/docs/etc. 
 

:hearts: Thank you!

andreslucena commented:
@alecslupu I think this is ready to be reviewed.  The failing spec of the generator I understand that are from using the edge branch, if you want I can change them so it pass, whatever approach you prefer.

ahukkanen commented:
Just a question that should we move the code to the archive repository?
https://github.com/decidim-archive/decidim-module-consultations

Of course the code will remain in the history of this repository but just thinking out loud when we have that archive existing as well.

fblupi commented:
> Just a question that should we move the code to the archive repository? https://github.com/decidim-archive/decidim-module-consultations
> 
> Of course the code will remain in the history of this repository but just thinking out loud when we have that archive existing as well.

I agree with @ahukkanen. Maybe some integrators prefer to keep using it and make it compatible with the redesign.

andreslucena commented:
> Just a question that should we move the code to the archive repository? [decidim-archive/decidim-module-consultations](https://github.com/decidim-archive/decidim-module-consultations)
> 
> Of course the code will remain in the history of this repository but just thinking out loud when we have that archive existing as well.

Regarding that README, as we're linking here saying that now lives here, but that's not the case, I propose that we change that explanation: 

```markdown

# decidim-module-consultations

In 2018, this module was integrated into the main repository: [decidim/decidim](https://github.com/decidim/decidim).

In 2021, we started developing an alternative module called `decidim-elections` and initiated the deprecation process for the original module. We added a callout in the admin panel for this module:

>  Consultations module will be deprecated in the near future. We’re working on the next cryptographically secure version called Votings.  

In 2023, as part of the Redesign process, we removed the code for this module from the main repository. Please refer to PR [#11171](https://github.com/decidim/decidim/pull/11171) for more details.

If you wish to maintain a new version of this module, you can recover it using git:

---shell
git clone https://github.com/decidim/decidim
git restore -s <COMMIT> decidim-consultations
---

If you're maintaining a version of this module, please share the URL of the git repository by [creating an issue on the decidim.org website repository](https://github.com/decidim/decidim.org) so that we can update the [Modules page](https://decidim.org/modules).
```

Only two things that need to change: the format for the code block (it should be "```shell" instead of "---shell") and the commit hash that we need to add after merging this PR.

alecslupu commented:
@andreslucena The PR is approved. 

Since this is quite a bug change, i would suggest to add some block text in the RELEASE_NOTES stating that we removed this module. 
Also, i recommend to add it to the archive.

andreslucena commented:
> Since this is quite a big change, i would suggest to add some block text in the RELEASE_NOTES stating that we removed this module.

Done at f12553bf875aec6bf3f4457417cb935ecd8d4e35 

Notice that I've added it as a "General note", and also I've added a placeholder title for the Redesign, as that's much more important than this change. 

> Also, i recommend to add it to the archive.

Done at https://github.com/decidim-archive/decidim-module-consultations/pull/89 

I've added the current last commit hash, we should update it there if we merge another PR before this one (or we can just ignore it as we probably will not introduce changes in `decidim-consultations`, but just in case 😄)

alecslupu commented:
Both specs failing in generators are complaining about not finding the consultations module.
